### ISSUE TITLE
Multiple global object keys

### ISSUE THREAD

**Issue by marcglasberg** [flutter/flutter#58116]
`GlobalKeys`, of course, are not to be abused. But sometimes they are necessary and I am finding problems with the restriction that only a single `GlobalObjectKey` must exist.

As a real example, you cannot use any widgets with a global key inside of a `ScrollablePositionedList` (see issue here: https://github.com/google/flutter.widgets/issues/137

More generally, if you have any animation or list that must display the same widgets at the same time in different positions in the screen, then you can't have a `GlobalKey` anywhere because it will be a duplicate. 

This problem would be solved with a key called `MultipleGlobalObjectKey` which would allow for repeating the key, and then instead of having a `currentState` getter it would have a `currentStates` getter, that would return a list of states instead of a single one. (then `currentState` could throw an error if there is more than one available state).

I believe what I am asking here is not a breaking change (but maybe I'm wrong?).

In other words, this would allow us to search the tree for all widgets (more than one) with some specific global key. 

I tried to create this key class myself, but the `mount` method does this:

```
if (key is GlobalKey) {
      key._register(this);
    }
```

So `MultipleGlobalObjectKey` would need to override the `_register` method and I can't do it outside of `framework.dart`.
